### ISSUE TITLE
ProgressBar log not clear

### ISSUE THREAD

**Issue by nawabhussain** [ddelange/mapply#8]
While using the following init method, the log _5it [04:10, 50.01s/it]_ is generated. I am not sure what the time (_04:10_) here specifies. Is it the time elapsed or the estimated time for completion of the process?

` mapply.init(
        n_workers=12,
        chunk_size=200,
        max_chunks_per_worker=12,
        progressbar=True
   )`

**Comment by ddelange** [comment]
Hi @nawabhussain,

The module uses a [`tqdm.auto.tqdm`](https://tqdm.github.io/docs/shortcuts/#tqdmauto) progress bar, which (upon completion) shows 
```
total_iterations [total_elapsed_time, average_time_per_iteration]
```
where one iteration is one chunk. So in your case, due to `chunk_size=200`, your dataframe was split up into five chunks, running on five workers (five iterations), which each took 50 seconds to process.

When running in an iPython Notebook, you should see a more fancy progress bar, with the same information.

Hope that helps :) otherwise, just drop another message!

**Comment by nawabhussain** [comment]
First of all, amazing work! Ran a process using this library and swifter, the difference between total time for both was huge. Just one question, would it not be possible to also show estimated time?

**Comment by ddelange** [comment]
Thank you :) I'm glad it's giving you a speed up!

If you are sure it makes sense to go multicore, you can also try `n_workers=-1, chunk_size=1`, where then the `max_chunks_per_worker` will be decisive for a sensible amount of chunks for your system (probably you have 12 physical cores?)

I will try to take a look whether [pimping](https://tqdm.github.io/docs/tqdm/#__init__) the `auto` progressbar is possible (the auto one works for Jupyter notebooks as well). The default `from tqdm import tqdm` will show an ETA for sure (but doesn't work for notebooks). I'm a bit surprised tbh that the auto one doesn't...

**Comment by nawabhussain** [comment]
I have 24 cores (Ryzen Threadripper 3960). Using n_workers=-1 gave me a MemoryError. I am not sure if this is a bug in the library or it was something on my end. I guess this is because not all the cores were free, for some reason. So I switched to 12 leaving a huge chunk for other processes. 

Thank you for the explanation about the progressbar. It was helpful. I will be waiting for the PR or if I get time soon, I will look into it myself and make a PR.

**Comment by ddelange** [comment]
Released [`0.1.7`](https://github.com/ddelange/mapply/releases/tag/0.1.7), which should start showing you an actual progress bar and an ETA. They were actually there before, which I accidentally changed with #5 😅 

Thanks again for reporting @nawabhussain!

**Comment by nawabhussain** [comment]
That was pretty quick !! Thanks again!
